service/kernel/oneapi/service_defines_oneapi.h

```diff
@@ -33,4 +33,18 @@
         }                                                                     \
     }
 
+#define DAAL_OVERFLOW_CHECK_BY_MULTIPLICATION_PTR(type, op1, op2, statusPtr)                               \
+    {                                                                                                      \
+        if (!(0 == (op1)) && !(0 == (op2)))                                                                \
+        {                                                                                                  \
+            type r = (op1) * (op2);                                                                        \
```

Comment:
with C++11 you can use keyword auto;

Reply:
It was decided in discussion with team members that explicit type is better here